index.bs

```diff
@@ -4957,14 +4957,13 @@ To <dfn>open a database connection</dfn> with |storageKey| which requested the [
         [=/connections=] in |openConnections| are
         [=connection/closed=].
 
-    1. Run [=upgrade a database=]
-        using |connection|, |version| and |request|.
+    1. Let |transaction| be the result of running [=upgrade a database=] with |connection|, |version| and |request|.
 
     1. If |connection| was [=connection/closed=],
         return a newly [=exception/created=]
         "{{AbortError}}" {{DOMException}} and abort these steps.
 
-    1. If the [=/upgrade transaction=] was aborted, run the steps
```

Comment:
I don't think you need to return a value from `upgrade a database` as the request's transaction will be set to the transaction, although only if it succeeded. If it was aborted, the request's transaction will be null and done flag will be false. So this can change to something like "If |request|'s [=request/transaction=] is null, run the steps to..."

Reply:
Thanks for this analysis, @evanstade. Agreed - I think this recommendation is sufficient. Even easier than what I recommended over in https://github.com/w3c/IndexedDB/issues/433#issuecomment-2512330086

Reply:
Good find @stelar7!  Upgrade transaction completion does [indeed reset the request's transaction to null](https://w3c.github.io/IndexedDB/#ref-for-request-transaction%E2%91%A2).  The [web platform tests also verify](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/web_tests/external/wpt/IndexedDB/upgrade-transaction-lifecycle-committed.any.js;drc=658153df0cd23ec41cb9f77e2689d342278d0ccd;l=53) this behavior.

Going back to the drawing board, maybe another potential alternative involves using the request's error, which the abort sets?

Reply:
Is there some reason we dont want to have the return? (what i feel is the simple solution)

Reply:
Good catch @stelar7.

I like Steve's idea.

> Is there some reason we dont want to have the return?

In general it seems preferable to work with the state that is already available, rather than making more state available to more of the system (especially when it is redundant). Returning the transaction changes its lifetime. There are fewer hypothetical side effects to reading state that already exists in the request.

Reply:
Updated to use the requests error 👍